include/mbedtls/aes.h

```diff
@@ -329,6 +329,57 @@ int mbedtls_aes_crypt_ctr( mbedtls_aes_context *ctx,
                        unsigned char *output );
 #endif /* MBEDTLS_CIPHER_MODE_CTR */
 
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+/**
+ * \brief      This function performs an AES-XTS encryption or decryption
+ *             operation for an entire XTS sector.
+ *
+ *             The key for tweak computation is separate from the key for AES
+ *             block encryption. So, two AES contexts must be supplied. XXX
+ *             Design review: It's a bit strange to have two AES ctxs.
+ *             Alternatives include a new key/ctx initialization function that
+ *             can handle the extra key needed by XTS, so long as that fits in
+ *             the existing context. This is sometimes handled as one big key
+ *             instead of two separate ones, key1 concat key2. Could store a
+ *             pointer in the ctx for the tweak's ctx, or could make a new ctx
+ *             for xts operations.
+ *
+ *             AES-XTS encrypts or decrypts blocks based on their location as
+ *             defined by a sector number. These must be provided by \p sector.
+ *
+ *             NIST SP 800-38E limits the maximum size of a sector to 2**20 AES
+ *             blocks. If the sector is larger than this, this function returns
+ *             #MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH.
+ *
+ * \param ctx          The AES context to use for AES block encryption or
+ *                     decryption.
+ * \param ctx_t        The AES context to use for tweak computation.
+ * \param mode         The AES operation: #MBEDTLS_AES_ENCRYPT or
+ *                     #MBEDTLS_AES_DECRYPT.
+ * \param length       The length of both an entire sector and the input data
+ *                     in bytes. The length must be at least 16 bytes. The
+ *                     length does not need to be a multiple of 16 bytes.
+ * \param sector       The sector address in byte array form. The least
+ *                     significant byte must be at sector[0]. The most
+ *                     significant byte must be at sector[15]. Array must be 16
+ *                     bytes in length.
+ * \param input        The buffer holding the input data.
+ * \param output       The buffer holding the output data.
+ *
+ * \return             \c 0 on success.
+ * \return             #MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH if length smaller
+ *                     than an AES block in size (16 bytes) or if the length is
+ *                     larger than 2**20 blocks.
+ */
+int mbedtls_aes_crypt_xts( mbedtls_aes_context *ctx,
+                           mbedtls_aes_context *ctx_t,
+                           int mode,
+                           size_t length,
+                           const unsigned char *sector,
```

Comment:
Since we know that this is 16 byte long, writing
``` diff
- const unsigned char *sector
+ const unsigned char sector[16]
```
would be slightly more expressive.

Reply:
I will change. The parameter array length is not enforced, so this is functionally not any different, but it does tell the programmer what we expect more clearly than just the documentation.